diff --git a/lib/constants2.js b/lib/constants2.js
index dc886f2496cd0c1333948c41559872b5fd9cb6e8..2878ef9f4ad2e2fae0e2c0d74b85258125f41c7f 100644
--- a/lib/constants2.js
+++ b/lib/constants2.js
@@ -27,12 +27,12 @@ const EXTENSIONS = ['.ts', '.tsx', '.cts', '.mts', '.js', '.jsx', '.cjs', '.mjs'
 const EXTENSIONS_WITHOUT_DOT = EXTENSIONS.map(ext => ext.slice(1));
 const EXTENSIONS_PATTERN = EXTENSIONS_WITHOUT_DOT.join(','); // Globs for finding source files, test files, and test utility files
 
-const ALL_FILES_GLOB = `**/{src,tests,__tests__}/**/*.{${EXTENSIONS_PATTERN}}`;
-const SOURCE_FILES_GLOB = `**/src/**/*.{${EXTENSIONS_PATTERN}}`;
-const TEST_FILES_GLOB = `**/{tests,__tests__}/**/*.test.{${EXTENSIONS_PATTERN}}`;
-const TEST_UTILS_GLOB = `**/{tests,__tests__}/**/*.{${EXTENSIONS_PATTERN}}`; // List of globs to find all test related files
+const ALL_FILES_GLOB = `**/{src,tests,__tests__}/**/!(*.d).{${EXTENSIONS_PATTERN}}`;
+const SOURCE_FILES_GLOB = `**/src/**/*!(*.d).{${EXTENSIONS_PATTERN}}`;
+const TEST_FILES_GLOB = `**/{tests,__tests__}/**/*.test!(*.d).{${EXTENSIONS_PATTERN}}`;
+const TEST_UTILS_GLOB = `**/{tests,__tests__}/**/!(*.d).{${EXTENSIONS_PATTERN}}`; // List of globs to find all test related files
 
-const TESTS_LIST = [TEST_FILES_GLOB, TEST_UTILS_GLOB, `test.{${EXTENSIONS_PATTERN}}`]; // Pattern of file extensions
+const TESTS_LIST = [TEST_FILES_GLOB, TEST_UTILS_GLOB, `**/*.test!(*.d).{${EXTENSIONS_PATTERN}}`]; // Pattern of file extensions
 
 const NON_JS_REGEX = '\\.(css|sass|scss|less|gif|png|jpg|jpeg|svg|gql|graphql|yml|yaml)$';
 const ALL_JS_REGEX = `\\.(${EXTENSIONS_WITHOUT_DOT.join('|')})$`; // Pattern to find all custom TypeScript paths
